contracts/axelar-gateway/src/auth.rs

```diff
@@ -235,3 +235,656 @@ fn validate_signers(env: &Env, weighted_signers: &WeightedSigners) -> Result<(),
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Most of these need to be converted into integration tests. Are you planning to do that in a separate PR?